### ISSUE TITLE
Publish/Download errors

### ISSUE THREAD

**Issue by gcy5s4ti** [eloquence/freeyourstuff.cc#43]
After going through over a thousand Quora answers (thanks), when hitting "publish" I get the error:

> There was a problem publishing your data. Sorry! :-( We got the following error message:
> JSON data could not be parsed, invalid syntax.
> You can help us out by filing a bug.

Hitting "download" doesn't help much either, as chromes gets a NETWORK_FAILED.
Maybe an issue in one of my 7 proxies?

UPDATE: I tested the "download" feature in another network and got the same error.

**Comment by eloquence** [comment]
Let me test the whole flow locally. I suspect it's a character not being correctly escaped or something like that. Thanks for the report and continuing patience! :)

**Comment by gcy5s4ti** [comment]
I know I used a russian (cyrillic) character in one of my answers

**Comment by eloquence** [comment]
There were a few separate issues here:
- Upload size in the web service was limited to 1MB. This file is a bit over that. That limit has now been significantly increased.
- There was a '</script>' tag (due to Quora's MathJax support) in one of the answers. Browsers interpret this as "script is over" even inside a string, which would have caused the data to display incorrectly even if the upload had succeeded. This will now be escaped.
- The way we handled downloads also choked on the size/complexity of the data. The extension now uses a more modern, reliable way of doing it.

I was able to download and upload your answers on my test system, so if you have any trouble I can send you a copy. But I'm curious if it works for you now. You know the drill - the extension will need to be updated to 0.3.5.
